go/libraries/doltcore/table/editor/pk_table_editor.go

```diff
@@ -97,6 +99,18 @@ type tableEditAccumulator struct {
 	affectedKeys map[hash.Hash]types.Value
 }
 
+// currentTableData represents the current persisted state of the table's row map. A row map with the set of added and
+// removed keys is equivalent to a row map that has had those keys applied through the edit accumulator. As a result,
+// queries can make use of the most up-to-date version of the data for the sake of comparisons (such as to find
+// duplicate keys on insert rather than flush) while the edit accumulator rebuilds the map in the background. All
+// functions assume that, at most, only one edit accumulator is flushing at any given time.
+type currentTableData struct {
+	mutex       *sync.RWMutex
+	rowData     types.Map
+	addedKeys   map[hash.Hash]types.Value
```

Comment:
Having these fields defined both here and in the tableEditAccumulator seems kind of fragile and easy to mess up. It's not always obvious to me when these need to be copied from `tea` and when they need to be replaced with empty maps.

Some ideas to make this clearer:

1) Add rowData to tableEditAccumulator, move all the currentTableData methods onto that struct
2) Make tableEditAccumulator responsible for creating / updating currentTableData objects

Reply:
Thinking it through, it does make more sense for `rowData` map to be in the `tea`. Solves some of my other synchronization problems as well.